File: WorldFeed/Server/src/WorldFeed.Streams/Model/AccountActivity/ActivityStreamDirectMessageConversationEventDTO.cs
﻿namespace WorldFeed.Streams.Model.AccountActivity
{
    using System;
    using Newtonsoft.Json;

    using WorldFeed.Common.JsonConverters;
    public class ActivityStreamTargetRecipientDTO
    {
        [JsonProperty("recipient_id")]
        public long RecipientId { get; set; }
    }

    public class ActivityStreamDirectMessageConversationEventDTO
    {
        [JsonProperty("created_timestamp")]
        [JsonConverter(typeof(JsonPropertyConverterRepository))]
        public DateTime CreatedAt { get; set; }

        [JsonProperty("sender_id")]
        public long SenderId { get; set; }

        [JsonProperty("target")]
        public ActivityStreamTargetRecipientDTO Target { get; set; }

        [JsonProperty("last_read_event_id")]
        public string LastReadEventId { get; set; }
    }
}

File: Sharebook/Client/src/app/streams/Model/AccountActivity/ActivityStreamDirectMessageConversationEventDTO.ts
export class ActivityStreamTargetRecipientDTO {
  // [JsonProperty("recipient_id")]
  public RecipientId: number;
}

export class ActivityStreamDirectMessageConversationEventDTO {
  // [JsonProperty("created_timestamp")]
  // [JsonConverter(typeof(JsonPropertyConverterRepository))]
  public CreatedAt: DateTimeOffset;

  // [JsonProperty("sender_id")]
  public SenderId: number;

  // [JsonProperty("target")]
  public Target: ActivityStreamTargetRecipientDTO;

  // [JsonProperty("last_read_event_id")]
  public LastReadEventId: string;
}
